### PR TITLE
improve internal types a little bit

### PR DESCRIPTION
**Pull request by talentlessguy**
as a bonus remove `flatten` util and use a native method instead

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -142,20 +148,8 @@ function anymatch(matchers: Matcher[], testString: string | undefined): boolean
   return matchPatterns(patterns, testString);
 }
 
-const flatten = (list: any[], result = []) => {
```
Comment (talentlessguy) [inline src/index.ts:145]: `Array#flat` is supported since Node v12

https://node.green/#ES2019-features-Array-prototype--flat--flatMap-

Comment (bluwy) [reply src/index.ts:145]: In the places where you call `.flat()`, I think you need to pass `.flat(Infinity)` to replicate the old behaviour. Though with that said, the types doesn't allow nested arrays in the first place so I'm not sure if flattening is needed, but maybe better to keep for compat?

Comment (talentlessguy) [reply src/index.ts:145]: well `any[]` can as well be `any[][]` so it doesn't differ

and `flatten` is an internal util that was only used once so I don't think setting infinite depth makes sense here

Comment (bluwy) [reply src/index.ts:145]: I don't mean `any[]`, but where it was being used and called here:

image

Arguably flattening with any level doesn't make sense given the types, but if we ignore the types and assume people could nest with any depth, then `.flat(Infinity)` matches the previous behaviour.

**Pull request merged**